### PR TITLE
Problem: unnecessary GetBalance in ante handlers

### PR DESCRIPTION
**Pull request by mmsqe**
reuse balance after verify account balance to avoid get again when check can transfer

benchmark need remove the skip [part](https://github.com/crypto-org-chain/ethermint/blob/fe3f4fd535d72e50e2628f44487d3159efb4d4ec/app/ante/eth.go#L72-L74) 

```
pkg: github.com/crypto-org-chain/cronos/v2/app
                                │  before.txt  │             after.txt             │
                                │    sec/op    │   sec/op     vs base              │
ERC20Transfer/memiavl-stm-1-12    870.5m ±  7%   866.1m ± 1%       ~ (p=0.699 n=6)
ERC20Transfer/memiavl-stm-8-12    180.7m ±  4%   174.9m ± 4%       ~ (p=0.065 n=6)
ERC20Transfer/memiavl-stm-16-12   205.6m ± 19%   198.5m ± 4%       ~ (p=0.180 n=6)
ERC20Transfer/memiavl-stm-32-12   243.7m ± 10%   247.1m ± 8%       ~ (p=0.818 n=6)
geomean                           297.9m         293.6m       -1.46%

                                │  before.txt  │             after.txt              │
                                │     B/op     │     B/op      vs base              │
ERC20Transfer/memiavl-stm-1-12    475.9Mi ± 0%   482.4Mi ± 0%  +1.36% (p=0.002 n=6)
ERC20Transfer/memiavl-stm-8-12    484.9Mi ± 0%   491.7Mi ± 0%  +1.39% (p=0.002 n=6)
ERC20Transfer/memiavl-stm-16-12   498.6Mi ± 1%   505.4Mi ± 0%  +1.36% (p=0.002 n=6)
ERC20Transfer/memiavl-stm-32-12   523.4Mi ± 1%   532.6Mi ± 1%  +1.77% (p=0.002 n=6)
geomean                           495.4Mi        502.7Mi       +1.47%

                                │ before.txt  │             after.txt             │
                                │  allocs/op  │  allocs/op   vs base              │
ERC20Transfer/memiavl-stm-1-12    5.998M ± 0%   6.023M ± 0%  +0.42% (p=0.002 n=6)
ERC20Transfer/memiavl-stm-8-12    6.127M ± 0%   6.153M ± 0%  +0.42% (p=0.002 n=6)
ERC20Transfer/memiavl-stm-16-12   6.347M ± 1%   6.365M ± 0%  +0.28% (p=0.041 n=6)
ERC20Transfer/memiavl-stm-32-12   6.758M ± 2%   6.801M ± 2%       ~ (p=0.240 n=6)
geomean                           6.301M        6.329M       +0.44%
```

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/ante/eth.go [hunk 1]
```diff
@@ -244,28 +260,25 @@ func CheckEthCanTransfer(
...
 		}
 	}
 
 	return nil
 }
 
-// canTransfer adapted the core.CanTransfer from go-ethereum
-func canTransfer(ctx sdk.Context, evmKeeper EVMKeeper, denom string, from common.Address, amount *big.Int) bool {
-	balance := evmKeeper.GetBalance(ctx, sdk.AccAddress(from.Bytes()), denom)
-	return balance.Cmp(amount) >= 0
```
Comment (mmsqe) [inline app/ante/eth.go:266]: balance could be nil, but there was no nil check before

**Comment by yihuang** [comment]
again, how does this affect the benchmark result, I guess the effect of this is even smaller than the account PR, which is almost unnoticeable in the benchmark result.

File: app/ante/eth.go (hunk 1 above)
Comment (yihuang) [reply app/ante/eth.go:266]: try to reproduce in integration test?

**Comment by mmsqe** [comment]
> again, how does this affect the benchmark result, I guess the effect of this is even smaller than the account PR, which is almost unnoticeable in the benchmark result.

Do you mean GetBalance is faster than GetAccount, no need to cache?

**Comment by yihuang** [comment]
> > again, how does this affect the benchmark result, I guess the effect of this is even smaller than the account PR, which is almost unnoticeable in the benchmark result.
> 
> Do you mean GetBalance is faster than GetAccount, no need to cache?

what do you think we are caching here? the raw bytes fetched from db is already cached by cachekv store, so what we are caching here is just to avoid the decoding step, if the decoding speed is similar to or even faster than the added map operations, then there's no point to add this cache, right?

**Pull request closed without merging**